internal/pkg/api/handleCheckin.go

```diff
@@ -1005,13 +1005,14 @@ func parseComponents(zlog zerolog.Logger, agent *model.Agent, req *CheckinReques
 	// Compare the deserialized meta structures and return the bytes to update if different
 	if !reflect.DeepEqual(reqComponents, agent.Components) {
 
+		reqComponentsJSON, _ := json.Marshal(*req.Components)
 		zlog.Trace().
-			RawJSON("oldComponents", agentComponentsJSON).
-			RawJSON("newComponents", *req.Components).
+			Str("oldComponents", string(agentComponentsJSON)).
+			Str("req.Components", string(reqComponentsJSON)).
 			Msg("local components data is not equal")
 
 		zlog.Info().
-			RawJSON("req.Components", *req.Components).
+			Str("req.Components", string(reqComponentsJSON)).
```

Comment:
Is there a reason for us to log the components JSON for every components change? Successful checkins will allow it to be queried from the .fleet-agents index at any time. This seems like something that would only be valuable to log whe there is an error.

At minimum this could be moved to the debug level, it doesn't make sense to log this at info by the unhealthy_reason at debug below, the unhealthy reason is much more interesting, but again it can be queried from .fleet-agents at any time.

Reply:
> Is there a reason for us to log the components JSON for every components change?

I'm not sure, honestly, I was just focusing on the mapping explosion. Someone from @elastic/fleet might be able to answer that.

The lines before already log the old and new components at trace level, so just removing them from this entry seems to be the best option, which will also avoid logging the same thing multiple times.